tests/ui/drop_forget_ref.stderr

```diff
@@ -15,6 +15,19 @@ note: argument has type &SomeStruct
 12 |     drop(&SomeStruct);
    |          ^^^^^^^^^^^
 
+warning: call to `std::mem::drop` with a value that implements Copy. Dropping a copy leaves the original intact.
```

Comment:
Should we make the drop_copy lint not work with references so that the drop_forget_ref lint can work?

In fact, it may make sense to do all four lints in a single pass.